modules/sfp_hostio.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_hostio
+# Purpose:      Host.io database query module
+#
+# Author:      Lev Trubach <leotrubach@gmail.com>
+#
+# Created:     2020-08-21
+# Copyright:   (c) Steve Micallef
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+import json
+
+from sflib import SpiderFootPlugin, SpiderFootEvent
+
+
+class sfp_hostio(SpiderFootPlugin):
+    """Host.io Query Module:Passive:Passive DNS:apikey:Obtain DNS information from host.io source."""
+
+    meta = {
+        "name": "Host.io Query Module",
+        "summary": "Obtain DNS information from host.io source.",
+        "flags": ["apikey"],
+        "useCases": ["Passive"],
+        "categories": ["Passive DNS"],
+        "dataSource": {
+            "website": "https://host.io",
+            "model": "FREE_AUTH_LIMITED",
+            "references": ["https://host.io/docs"],
+            "apiKeyInstructions": [
+                "Visit https://host.io/signup",
+                "Register a free account",
+                "Visit https://host.io/dashboard and use the authentication token provided",
+            ],
+            "favIcon": "https://host.io/static/images/hostio/favicon.png?v2",
+            "logo": "https://host.io/static/images/hostio/favicon.png?v2",  # Seems like they embed it as SVG
+            "description": "We collect data on every known domain name, from every TLD, and update it every month. "
+            "Our data includes DNS records and website data for each of the domains."
+            "We process terabytes of data and summarize it to produce our final results. "
+            "Browse through our site to see backlinks, redirects, server details or IP address "
+            "and hosting provider details courtesy of IPinfo.io.",
+        },
+    }
+
+    opts = {
+        "api_key": "",
+        "checkcohosts": True,
+        "checkaffiliates": True,
+        "maxcohost": 100,
+        "verify": True,
+    }
+
+    optdescs = {
+        "api_key": "Host.io API Key.",
+        "checkcohosts": "Check co-hosted sites?",
+        "checkaffiliates": "Check affiliates?",
+        "maxcohost": "Stop reporting co-hosted sites after this many are found, as it would likely indicate web hosting.",
+        "verify": "Verify that any hostnames found on the target domain still resolve?",
+    }
+
+    errorState = False
+
+    def setup(self, sfc, userOpts=None):
+        if userOpts is None:
+            userOpts = {}
+        self.sf = sfc
+        self.results = self.tempStorage()
+        self.opts.update(userOpts)
+
+    def watchedEvents(self):
+        return [
+            "AFFILIATE_DOMAIN_NAME",
+            "CO_HOSTED_SITE_DOMAIN",
+            "DOMAIN_NAME",
+        ]
+
+    # What events this module produces
+    def producedEvents(self):
```

Comment:
The module could get more from their API, like EMAILADDR, WEB_ANALYTICS_ID and WEBSERVER_TECHNOLOGY.